File: frontend/src/model/history.ts
export type history = {
  articleId: number;
  friendId: string;
  date: string;
  type: string;
  amount: number;
  item: string;
  given: boolean;
  errorMsg: string;
};

export type historyDetail = history & { detail: string; image: string };

export type historyList = {
  list: history[];
};

File: server/src/main/java/com/galaxy/memorit/history/domain/entity/History.java
package com.galaxy.memorit.history.domain.entity;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class History {
	private Long id;
	private String userId;
	private String friendId;
	private LocalDate date;
	private String type;
	private Integer amount;
	private String item;
	private String detail;
	private String image;
	private boolean given;
}
